DEV/org.openl.rules/src/org/openl/rules/helpers/RulesUtils.java

```diff
@@ -3245,45 +3236,14 @@ public static Object staticField(Object instance, String fieldName) {
         return null;
     }
 
-    @SuppressWarnings("unchecked")
+    @MethodSearchTuner(wrapper = CopyMethodCallerWrapper.class)
     public static <T> T copy(T origin) {
         // Create new instance every time because of the issue with updating a module in WebStudio.
         // It is needed to investigate class loading/unloading mechanism.
         if (origin == null) {
             return null;
         }
-
-        final Class<?> clazz = origin.getClass();
-        if (clazz.isArray() && Serializable.class
-            .isAssignableFrom(clazz.getComponentType()) || !clazz.isArray() && origin instanceof Serializable) {
-            ObjectOutputStream oos = null;
-            ObjectInputStream ois = null;
-            try {
-                ByteArrayOutputStream baos = new ByteArrayOutputStream(512);
-                oos = new ObjectOutputStream(baos);
-                oos.writeObject(origin);
-
-                ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
-                ois = new ObjectInputStream(bais) {
-                    @Override
-                    protected Class<?> resolveClass(ObjectStreamClass desc) throws IOException, ClassNotFoundException {
-                        try {
-                            return Class.forName(desc.getName(), false, clazz.getClassLoader());
-                        } catch (ClassNotFoundException cnfe) {
-                            // delegate to super class loader which can resolve primitives
-                            return super.resolveClass(desc);
-                        }
-                    }
-                };
-                return (T) ois.readObject();
-            } catch (Exception unused) {
-                // OK lets use cloner
-            } finally {
-                IOUtils.closeQuietly(oos);
-                IOUtils.closeQuietly(ois);
-            }
-        }
-        // FIXME: Needless memory consumption - no needs to create 'cloner' instance every time.
-        return new OpenLArgumentsCloner().deepClone(origin);
+        CopyMethodDetails copyMethodDetails = (CopyMethodDetails) MethodDetailsMethodCaller.getMethodDetails();
+        return copyMethodDetails.getCloner().deepClone(origin);
```

Comment:
Can we use `new OpenLCloner()` directly without wrapping into `CopyMethodDetails` ?

Reply:
No, we can't. Please, read deleted comment. I mean that we can create a new instance on each call but cloner builds cache and work not effective on the first call. This solution improves the performance of the clone method and doesn't have memory leak problems.